obfuscator.go

```diff
@@ -0,0 +1,76 @@
+package sqllexer
+
+import (
+	"regexp"
+	"strings"
+)
+
+type SQLObfuscatorConfig struct {
+	ReplaceDigits    bool
+	DollarQuotedFunc bool
+}
+
+type SQLObfuscator struct {
+	config *SQLObfuscatorConfig
+}
+
+func NewSQLObfuscator(config *SQLObfuscatorConfig) *SQLObfuscator {
+	return &SQLObfuscator{config: config}
+}
+
+const (
+	StringPlaceholder = "?"
+	NumberPlaceholder = "?"
+)
+
+// Obfuscate takes an input SQL string and returns an obfuscated SQL string.
+// The obfuscator replaces all literal values with a single placeholder
+func (o *SQLObfuscator) Obfuscate(input string) string {
+	var obfuscatedSQL string
```

Comment:
Given how we might have a long sql statement and many tokens, I think it would make sense to use [strings.Builder](https://pkg.go.dev/strings#Builder) for this to make this code more efficient.